### PR TITLE
[SG-33093] Enable accessibilityAudit on `client/web/src/integration/batches.test.ts`

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Description

- Enabling accessibility audit on `batches` page in its integration test

## Refs

[SG Issue](https://github.com/sourcegraph/sourcegraph/issues/33093)
[Gitstart Ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-33093)

## Test plan

Run the integration test `yarn test-integration:base client/web/src/integration/batches.test.ts`

## Success criteria

- All integration tests with percy snapshots have `accessibilityAudit` enabled

## App preview:
- [Link](https://sg-web-contractors-sg-33093-1-1.onrender.com)

### CODE REVIEW

**Comment by gitstart-sourcegraph** [comment]
@umpox @valerybugakov  could you review the first PR for `accessibilityAudit`? Would need your suggestions on some `a11y-ignore` usages (already left comments for issues)? 

BTW it seems that we know another issue of current `wildcard/Menu` with Reach Menu + Popover, it's about Popover removes `MenuList` from DOM, it causes `aria-controls` attribute in `MenuButton` invalid

**Comment by umpox** [comment]
@gitstart-sourcegraph The `aria-controls` issue is interesting. We could work around this by using `isExpanded` from `Menu` to only add `aria-controls` when visible. I think this is OK though, it seems from reading this issue: https://github.com/w3c/aria/issues/995 that `aria-controls` has relatively low support. We should be able to make these components accessible without it

File: client/web/src/components/AppRouterContainer/AppRouterContainer.tsx [hunk 1]
```diff
@@ -14,10 +14,11 @@ export const AppRouterContainer: React.FunctionComponent<AppRouterContainerProps
 }) => (
     <ElementScroller scrollKey="app-router-container">
         {/*
-            Data Layout data attribute is used to get access to the main layout scroll element (the div element below).
+            - Data Layout data attribute is used to get access to the main layout scroll element (the div element below).
             from child levels in order to handle or react on this container scroll or other important events.
+            - `role="main"` to fix rule: "landmark-one-main" (Document should have one main landmark)
          */}
-        <div data-layout={true} className={classNames(styles.appRouterContainer, className)} {...rest}>
+        <div role="main" data-layout={true} className={classNames(styles.appRouterContainer, className)} {...rest}>
```
Comment (umpox) [inline client/web/src/components/AppRouterContainer/AppRouterContainer.tsx:21]:
```suggestion
        <main data-layout={true} className={classNames(styles.appRouterContainer, className)} {...rest}>
```

I think this should be equivalent

File: client/web/src/enterprise/batches/list/GettingStarted.tsx [hunk 2]
```diff
@@ -149,7 +149,13 @@ export const GettingStarted: React.FunctionComponent<GettingStartedProps> = ({ f
                             View the product marketing page for a high-level overview of benefits and customer use
                             cases.
                         </p>
-                        <Link to="https://about.sourcegraph.com/batch-changes" rel="noopener">
+                        {/*
+                            a11y-ignore
+                            Rule: "color-contrast" (Elements must have sufficient color contrast)
+                            There are a lot of `Link` usages in `--color-bg-2=--gray-03` container, to fix this issue
+                            we can use `--gray-02` or find another color for `--link-color`
```
Comment (umpox) [inline client/web/src/enterprise/batches/list/GettingStarted.tsx:156]:
```suggestion
                            There are a lot of `Link` usages in `--color-bg-2=--gray-03` container, to fix this issue
                            we can use `--gray-02` or find another color for `--link-color`
```

@gitstart-sourcegraph Lets just link to a GitHub issue for these contrast issues. I think we will encounter quite a few of them. We might not want to address all of these as the WCAG color contrast algorithm is changing to be more accurate in the next version of the specification (WCAG 3). Let's leave a comment with a link to this issue wherever we have this problem (and it doesn't appear to be a genuine bug) .

```suggestion
                            GitHub issue: https://github.com/sourcegraph/sourcegraph/issues/33343
```

File: client/web/src/nav/StatusMessagesNavItem.tsx [hunk 3]
```diff
@@ -475,7 +475,12 @@ export class StatusMessagesNavItem extends React.PureComponent<Props, State> {
 
         return (
             <Popover isOpen={this.state.isOpen} onOpenChange={event => this.setState({ isOpen: event.isOpen })}>
-                <PopoverTrigger className="nav-link py-0 px-0 percy-hide chromatic-ignore" as={Button} variant="link">
+                <PopoverTrigger
+                    className="nav-link py-0 px-0 percy-hide chromatic-ignore"
+                    as={Button}
+                    variant="link"
+                    aria-label={this.state.isOpen ? 'Hide tasks status message' : 'Show tasks status message'}
```
Comment (umpox) [inline client/web/src/nav/StatusMessagesNavItem.tsx:482]:
```suggestion
                    aria-label={this.state.isOpen ? 'Hide status messages' : 'Show status messages'}
```

**Review by umpox — CHANGES_REQUESTED** [review]
Thanks! Left some some comments

File: client/wildcard/src/components/Menu/MenuButton.tsx [hunk 4]
```diff
@@ -14,11 +15,21 @@ export type MenuButtonProps = Omit<ButtonProps, 'as'>
...
-    </ReachMenuButton>
-)) as ForwardReferenceComponent<'button', MenuButtonProps>
+export const MenuButton = React.forwardRef(({ children, id, ...props }, reference) => {
+    // To fix rule: "duplicate-id-active"
+    // Document has active elements with the same id attribute: menu-button--menu
+    const uniqueMenuId = useMemo(() => id ?? uniqueId('menu-button-'), [id])
+
+    // aria-controls="" to fix "aria-valid-attr-value" since current version of
+    // Popover removes contents from DOM, it causes `aria-controls="menu-*"` invalid
+    // since `menu-*` doesn't exit on DOM
```
Comment (umpox) [inline client/wildcard/src/components/Menu/MenuButton.tsx:25]:
```suggestion
    // We unset aria-controls as it causes accessibility issues if the Menu is not yet rendered in the DOM.
    // aria-controls has low support across screen readers so this shouldn't be an issue: https://github.com/w3c/aria/issues/995
```

**Comment by gitstart-sourcegraph** [comment]
@umpox could you help to check percy snapshots reports? It seems that all snapshots from this PR is new

File: client/web/src/components/AppRouterContainer/AppRouterContainer.tsx [hunk 5]
```diff
@@ -14,11 +14,12 @@ export const AppRouterContainer: React.FunctionComponent<AppRouterContainerProps
 }) => (
     <ElementScroller scrollKey="app-router-container">
         {/*
-            Data Layout data attribute is used to get access to the main layout scroll element (the div element below).
+            - Data Layout data attribute is used to get access to the main layout scroll element (the div element below).
             from child levels in order to handle or react on this container scroll or other important events.
+            - `role="main"` to fix rule: "landmark-one-main" (Document should have one main landmark)
```
Comment (umpox) [inline client/web/src/components/AppRouterContainer/AppRouterContainer.tsx:19]:
```suggestion
```

We can remove this now

File: client/wildcard/src/components/PageHeader/PageHeader.tsx [hunk 6]
```diff
@@ -51,8 +51,13 @@ export const PageHeader: React.FunctionComponent<Props> = ({
...
     }
 
+    /// a11y-ignore
+    /// Rule: "landmark-banner-is-top-level" (Banner landmark should not be contained in another landmark)
+    // We should think about stop using `header` here since we couldn't make it be be top-level
+    // with current usages
+
     return (
-        <header className={classNames(styles.container, className)}>
+        <header className={classNames(styles.container, className, 'a11y-ignore')}>
```
Comment (umpox) [inline client/wildcard/src/components/PageHeader/PageHeader.tsx:60]:
```suggestion
        <div className={classNames(styles.container, className)}>
```

Let's just keep this as a `div`. Header may be the wrong element here as we can't control where this component is used

**Review by umpox — APPROVED** [review]
Two small comments, rest looks good! I think the Percy changes aren't related to this PR, but because we're currently having some issues with Percy builds on main. Will look into it and have re-ran the build to check if that fixes

**Comment by gitstart-sourcegraph** [comment]
Thanks @valerybugakov  @umpox  all checks are good except percy check. Should we merge this PR or wait for the fixes?

**Pull request merged**